### ISSUE TITLE
Training DETR-RT model yields 0 metrics

### ISSUE THREAD

**Issue by ruiz-manuel** [ultralytics/ultralytics#20271]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have a dataset that I have used multiple times to train different YOLO models using Ultralytics. When I try to use the same data to train a detr-rt model, I get 0 in all metrics after 1 epoch of training. The dataset is big (thousands of images). I don't expect it to be good after just one epoch but not flat 0. Val images show no detections whatsoever.

If I train yolov11 for just one epoch I get about 85% MAP50.

Something is wrong.

What I have tried:
-set optimizer to AdamW  (so that lr is 0.01 and not 0.00001 as with auto)
-change imgsz
-change batch size 8 or 16
-deleting cached labels before training

Here's an example command:
yolo detect train data=data/custom-data.yaml model=rtdetr-l.pt epochs=1  batch=8 

![Image](https://github.com/user-attachments/assets/9a509f29-eb51-4bee-a1fa-bf61cf681e99)

The cls_loss seems to be effectively going down, but too much. It is almost 0 quickly. What could be wrong?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ruiz-manuel, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response. An Ultralytics engineer will review your issue and assist you soon! 😊

**Comment by Y-T-G** [comment]
You can try this config

https://github.com/ultralytics/ultralytics/issues/19530#issuecomment-2702930126

**Comment by ruiz-manuel** [comment]
It works now! Not sure what parameter specifically makes it work. In trains very slow though, not sure how I can reach the same metrics I was getting with YOLOv11.

My cls train loss makes something really weird, maybe gain is too high? (this is with dfl, cls and box all 1.0)

![Image](https://github.com/user-attachments/assets/28582f20-31e8-4614-8ec1-372679e0bb3f)

**Comment by Y-T-G** [comment]
You can try increasing lr to 0.0001

**Comment by Y-T-G** [comment]
Also try `nbs=1`

**Comment by ruiz-manuel** [comment]
"Nominal batch size for normalization of loss."

> Also try `nbs=1`

Is this recommendation for training faster or better accuracy?

**Comment by Y-T-G** [comment]
Accuracy
